Updating SEO articles (#5515)
PostHog/posthog.com: contents/blog/best-open-source-analytics-tools.md

  - Comparisons
---

There's no shortage of high-quality open-source analytics tools. In this guide we've split them into two categories:

- [Analytics tools](#the-best-open-source-analytics-tools) that collect data and help you understand user behavior. This includes product analytics platforms, but also web analytics, session recording, and AB testing tools.

> - [Dashboarding tools](#the-best-open-source-dashboard-and-bi-tools) that connect to data sources (e.g. a data warehouse) for building dashboards, insights, interactive visualizations, and real-time data logging.

Reviewer: Should we write an article for this instead? Both open source visualization tools and open source dashboard tools get solid (~400/m) search traffic.

> - **Mar 14, 2022:** Added Umami; updates to copy and screenshots

## The best open-source analytics tools

This section is dedicated to tools that actively collect data on user behavior. These range from the very simple, such as the privacy-focused Plausible, to broad analytics tools like Matomo.

### PostHog
![PostHog - best open source analytics tools](../images/blog/gdpr-compliant-analytics/posthog-gdpr-compliant.png)

- **GitHub Stars:** 11.2k
- **Language(s):** TypeScript (53%) and Python (42%)
- **Alternative to:** Mixpanel, Amplitude, Hotjar, FullStory, LaunchDarkly
- [PostHog on GitHub](https://github.com/PostHog/posthog)

[PostHog](https://posthog.com/) is an all-in-one platform designed to give engineers and product managers a complete view of user behavior. It's built on the highly-scalable ClickHouse OLAP database.

Features include a comprehensive product analytics suite (e.g. trend analysis, dashboards, funnels, path analysis, retention tracking etc.), feature flags for feature control, experimentation (A/B and multivariate testing), session recording, heatmaps and [more](/product).

> It’s entirely self-serve, with the ability to auto-capture event information out of the box. PostHog can also be self-hosted on your existing infrastructure using Docker Compose, making it ideal if you don't want to share data with third parties. 

Reviewer: ```suggestion
It’s entirely self-serve, with the ability to autocapture event information out of the box. PostHog can also be self-hosted on your existing infrastructure using Docker Compose, making it ideal if you don't want to share data with third parties. 
```

#### Open source license and monetization
Plausible Analytics is available under a AGPLv3 license and can be self-hosted via Docker Compose. Pricing for its hosted version starts at €9 per month for 10,000 monthly pageviews.

> **Also consider:** Plausible is one of a large subset of privacy-first analytics products that have proven popular in the wake of GDPR and less permissive attitudes to user tracking. [Umami](https://umami.is/) is another popular, open-source product in this space. Read our [GDPR analytics guide](/blog/best-gdpr-compliant-analytics-tools) for more options.

### Umami

![umami analytics](../images/blog/open-source-analytics-tools/umami.png)

- **GitHub Stars:** 14.9k
- **Languages:** 91% JavaScript
- **Alternative to:** Google Analytics
- [Umami on GitHub](https://github.com/umami-software/umami)

> [Umami](https://umami.is/) is another popular privacy-first website analytics tool. Like Plausible, it only tracks basic website analytics, but it doesn't collect any person information and is extremely fast. You can also track basic events, such button clicks, using CSS classes or JavaScript. Unlike Plausible, it doesn't currently offer a cloud-hosted option, though one is in development. 

Reviewer: ```suggestion
[Umami](https://umami.is/) is another popular privacy-first website analytics tool. Like Plausible, it tracks basic website analytics, doesn't collect any personal information, and is extremely fast. You can also track basic events, such button clicks, using CSS classes or JavaScript. Unlike Plausible, it doesn't currently offer a cloud-hosted option, though one is in beta. 
```

### Grafana
![Grafana - open source analytics tools](../images/blog/open-source-analytics-tools/grafana-screenshot.png)

- **GitHub Stars:** 54.4k
- **Languages:** TypeScript (57%), Go (37%)
- **Alternative to:** Datadog, Dynatrace 
- [Grafana on Github](https://github.com/grafana/grafana)

> [Grafana](https://grafana.com/products/cloud/) is an open-source data observability. It focuses on interactive visualization, providing a wide variety of charts, graphs and alerts that can be connected to a wide range of online data sources. As a result, it’s a powerful system for monitoring and analyzing data in real-time.  

Reviewer: ```suggestion
[Grafana](https://grafana.com/products/cloud/) is an open-source data observability platform. It focuses on interactive visualization, providing a wide variety of charts, graphs and alerts that can be connected to a wide range of online data sources. As a result, it’s a powerful system for monitoring and analyzing data in real-time.  
```
